src/braket/parametric/free_parameter_expression.py

```diff
@@ -30,7 +40,11 @@ class FreeParameterExpression:
     present will NOT run. Values must be substituted prior to execution.
     """
 
-    def __init__(self, expression: Union[FreeParameterExpression, Number, Expr, str]):
+    def __init__(
+        self,
+        expression: Union[FreeParameterExpression, Number, Expr, str],
+        _type: Optional[ClassicalType] = None,
```

Comment:
Do we expect this to be a private optional param? If not, we should move this to be `type` instead of `_type`